main.go

```diff
@@ -148,11 +155,52 @@ func main() {
 	// ********************************************************************************
 	server := grpc.NewServer(grpc.Creds(grpcfd.TransportCredentials(credentials.NewTLS(tlsconfig.MTLSServerConfig(source, source, tlsconfig.AuthorizeAny())))))
 	endpoint.Register(server)
-	srvErrCh := grpcutils.ListenAndServe(ctx, &config.ListenOn, server)
+	tmpDir, err := ioutil.TempDir("", "forwarder-")
+	if err != nil {
+		logrus.Fatalf("error getting x509 svid: %+v", err)
+	}
+	defer func(tmpDir string) { _ = os.Remove(tmpDir) }(tmpDir)
+	listenOn := &(url.URL{Scheme: "unix", Path: filepath.Join(tmpDir, "socket")})
+	srvErrCh := grpcutils.ListenAndServe(ctx, listenOn, server)
 	exitOnErr(ctx, cancel, srvErrCh)
+
+	// ********************************************************************************
+	log.Entry(ctx).Infof("executing phase 6: register xconnectns with the registry (time since start: %s)", time.Since(starttime))
+	// ********************************************************************************
+	registryCreds := credentials.NewTLS(tlsconfig.MTLSClientConfig(source, source, tlsconfig.AuthorizeAny()))
+	registryCreds = grpcfd.TransportCredentials(registryCreds)
+	registryCC, err := grpc.DialContext(ctx,
+		config.ConnectTo.String(),
+		grpc.WithTransportCredentials(registryCreds),
+		grpc.WithBlock(),
+	)
+	if err != nil {
+		log.Entry(ctx).Fatalf("failed to connect to registry: %+v", err)
+	}
+
+	registryClient := registrychain.NewNetworkServiceEndpointRegistryClient(
+		registrysendfd.NewNetworkServiceEndpointRegistryClient(),
```

Comment:
Is here a missed `interpose client` chain element?

Reply:
Why would the interpose client chain element be needed there?

Reply:
Interpose client element makes registration a valid for the interpose server registry